src/main/java/com/uploadcare/api/RequestHelper.java

```diff
@@ -88,12 +88,14 @@ public String makeSignature(HttpUriRequest request, String date, String requestB
         }
 
         StringBuilder sb = new StringBuilder();
+        String uriString = request.getURI().toString();
+        int uriStartIndex = uriString.indexOf(Urls.API_BASE) + Urls.API_BASE.length();
+        String uriWithoutBase = uriString.substring(uriStartIndex);
         sb.append(request.getMethod())
                 .append("\n").append(requestBodyMD5)
                 .append("\n").append(JSON_CONTENT_TYPE)
                 .append("\n").append(date)
-                .append("\n").append(request.getURI().getPath());
-
+                .append("\n").append(uriWithoutBase);
```

Comment:
what happens if base partly includes path, e.g.
`my.example.com/api/`?

looking at pyuploadcare's code, it will break :)

Reply:
`public static final String API_BASE = "https://api.uploadcare.com";`
 is final static variable, it won't change or include anything else, also indexOf returns first appearance in any fantastic case...also there is just no cases when it's not there, all tests were passed successfully which included all REST/UPLOAD API requests.

Reply:
but what IF a user needs to change the base?
I know that this is out of scope, but if sometimes we'll add this, this particular function will stop working (seemingly without connection).

Reply:
This is hardcoded REST API base url, you can change it only in Urls.API_BASE variable, it is only used for REST API requests, it will still work the same exact way if you change it to anything else. Users cannot change it, if they do they won't be able to use REST API....I don't understand what is the problem here.

Reply:
The problem is in naive approach used in this method that either will result in bugs when we'll allow users to change the base or we'll have to rewrite it.
I believe that rewriting it now is better.

Changing API base is legitimate use case and frankly I'm surprised that it's not a part of the library yet.
@rsedykh please make sure that we have this requirement for the future.

Reply:
@dmitry-mukhin refactored in d5ae72b , removed any dependency on `Urls.API_BASE` variable. This is maximum that I can do with existing code. Should fix all your concerns above. @rsedykh